testutil/gittest/gittest.go

```diff
@@ -97,6 +105,97 @@ func NewServer(fs billy.Filesystem) http.Handler {
 	return mux
 }
 
+func NewServerSSH(t *testing.T, fs billy.Filesystem, pubkeys ...gossh.PublicKey) *transport.Endpoint {
+	t.Helper()
+
+	l, err := net.Listen("tcp", "localhost:0")
+	require.NoError(t, err)
+	t.Cleanup(func() { _ = l.Close() })
+
+	srvOpts := []ssh.Option{
+		ssh.PublicKeyAuth(func(ctx ssh.Context, key ssh.PublicKey) bool {
+			for _, pk := range pubkeys {
+				if ssh.KeysEqual(pk, key) {
+					return true
+				}
+			}
+			return false
+		}),
+	}
+
+	done := make(chan struct{}, 1)
+	go func() {
```

Comment:
I guess this is some go routine stuff right?

Reply:
Yeah we don't want `ssh.Serve()` to block execution but we want to wait for it to finish before continuing.

A common pattern here is to use use the fact that recieving on a closed channel will always immediately return the zero value for the channel. So you'l often see stuff like this:

```
done := make(chan struct{})
go func() {
  someBlockingFunction()
  close(done)
}()

<-done // waits for someBlockingFunction to finish
```